device/iot-device-client/src/test/java/com/microsoft/azure/sdk/iot/device/edge/MethodResultTest.java

```diff
@@ -25,7 +25,7 @@ public void constructorParsesJsonAndGettersWork()
         MethodResult result = new MethodResult(json);
 
         //assert
-        assertEquals("somePayload", result.getPayload());
+        assertEquals("somePayload", result.getPayloadAsJsonString());
```

Comment:
Let's have tests for simple types (as you have here), arrays, and a custom object (aka a map).